terraform/modules/app-ecs-albs/main.tf

```diff
@@ -352,12 +443,189 @@ resource "aws_lb_listener_rule" "prom_listener_https" {
+#   }
+# }
+
+resource "aws_security_group" "alertmanager_alb" {
+  name        = "${var.stack_name}-alertmanager-alb"
+  description = "Access to alertmanager ALB (${var.stack_name})"
+  vpc_id      = "${local.vpc_id}"
+}
+
+# Unlike prometheus, alertmanager is IP restricted by security group
+# here
+resource "aws_security_group_rule" "alerts_allow_http" {
+  security_group_id = "${aws_security_group.alertmanager_alb.id}"
+  type              = "ingress"
+  from_port         = 80
+  to_port           = 80
+  protocol          = "tcp"
+  cidr_blocks       = ["${var.alerts_allowed_cidrs}"]
+}
+
+resource "aws_security_group_rule" "alerts_allow_https" {
+  security_group_id = "${aws_security_group.alertmanager_alb.id}"
+  type              = "ingress"
+  from_port         = 443
+  to_port           = 443
+  protocol          = "tcp"
+  cidr_blocks       = ["${var.alerts_allowed_cidrs}"]
+}
+
+resource "aws_lb" "alertmanager_alb" {
+  name               = "${var.stack_name}-alertmanager-alb"
+  internal           = false
+  load_balancer_type = "application"
+
+  security_groups = [
+    "${data.terraform_remote_state.infra_security_groups.monitoring_external_sg_id}",
+    "${aws_security_group.alertmanager_alb.id}",
+  ]
+
+  subnets = [
+    "${var.subnets}",
+  ]
+
+  tags = "${merge(
+    local.default_tags,
+    var.additional_tags,
+    map("Stackname", "${var.stack_name}"),
+    map("Name", "${var.stack_name}-alertmanager-alb")
+  )}"
+}
+
+resource "aws_lb_listener" "alertmanager_listener_http" {
+  load_balancer_arn = "${aws_lb.alertmanager_alb.arn}"
+  port              = "80"
+  protocol          = "HTTP"
+
+  default_action {
+    type = "redirect"
+
+    redirect {
+      port        = "443"
+      protocol    = "HTTPS"
+      status_code = "HTTP_301"
+    }
+  }
+}
+
+resource "aws_lb_listener" "alertmanager_listener_https" {
+  load_balancer_arn = "${aws_lb.alertmanager_alb.arn}"
+  port              = "443"
+  protocol          = "HTTPS"
+  ssl_policy        = "ELBSecurityPolicy-TLS-1-2-2017-01"
+  certificate_arn   = "${aws_acm_certificate_validation.alertmanager_cert.certificate_arn}"
+
+  default_action {
+    type = "fixed-response"
+
+    fixed_response {
+      content_type = "text/plain"
+      message_body = "Not found"
+      status_code  = "404"
+    }
+  }
+}
+
+resource "aws_lb_listener_rule" "alerts_listener_https" {
+  count = "${var.alertmanager_count}"
+
+  listener_arn = "${aws_lb_listener.alertmanager_listener_https.arn}"
+  priority     = "${100 + count.index}"
+
+  action {
+    type             = "forward"
+    target_group_arn = "${element(aws_lb_target_group.alertmanager_tg.*.arn, count.index)}"
+  }
+
+  condition {
+    field = "host-header"
+
+    values = [
+      "alerts-${count.index + 1}.*",
+    ]
+  }
+}
+
+resource "aws_lb_target_group" "alertmanager_tg" {
+  count = "${var.alertmanager_count}"
+
+  name                 = "${var.stack_name}-alerts-${count.index +1}-tg"
+  port                 = 80
```

Comment:
If I understand this correctly, our target group will send traffic to our alertmanager on port 80? But at the moment we likely want this to go to port 9093 which our ECS instances expose for alertmanager? I might have misunderstood something here though

Reply:
🤔 you know, I’m not at all sure though I think you are probably right